src/integrations/bypass-post-meta-integration.php

```diff
@@ -0,0 +1,161 @@
+	 *                              If specified, only update existing metadata entries with
+	 *                              this value. Otherwise, update all entries.     *
+	 *
+	 * @return bool
+	 */
+	public function filter_update_post_meta( $check, $object_id, $meta_key, $meta_value, $prev_value ) {
+		return $this->filter_update_meta( $check, $object_id, $meta_key, $meta_value, 'post' );
+	}
+
+	/**
+	 * Filter updating the post meta.
+	 *
+	 * @param null|bool $check      Whether to allow updating metadata for the given type.
+	 * @param int       $object_id  ID of the object metadata is for.
+	 * @param string    $meta_key   Metadata key.
+	 * @param mixed     $meta_value Metadata value. Must be serializable if non-scalar.
+	 * @param mixed     $prev_value Optional. Previous value to check before updating.
+	 *                              If specified, only update existing metadata entries with
+	 *                              this value. Otherwise, update all entries.     *
+	 *
+	 * @return bool
+	 */
+	public function filter_update_term_meta( $check, $object_id, $meta_key, $meta_value, $prev_value ) {
+		return $this->filter_update_meta( $check, $object_id, $meta_key, $meta_value, 'term' );
+	}
+
+	/**
+	 * Filter updating the post meta.
+	 *
+	 * @param null|bool $check      Whether to allow updating metadata for the given type.
+	 * @param int       $object_id  ID of the object metadata is for.
+	 * @param string    $meta_key   Metadata key.
+	 * @param mixed     $meta_value Metadata value. Must be serializable if non-scalar.
+	 * @param mixed     $prev_value Optional. Previous value to check before updating.
+	 *                              If specified, only update existing metadata entries with
+	 *                              this value. Otherwise, update all entries.     *
+	 *
+	 * @return bool
+	 */
+	public function filter_update_user_meta( $check, $object_id, $meta_key, $meta_value, $prev_value ) {
+		return $this->filter_update_meta( $check, $object_id, $meta_key, $meta_value, 'user' );
+	}
+
+	/**
+	 * Filter updating a meta field.
+	 *
+	 * @param null|bool $check      Whether to allow updating metadata for the given type.
+	 * @param int       $object_id  ID of the object metadata is for.
+	 * @param string    $meta_key   Metadata key.
+	 * @param mixed     $meta_value Metadata value. Must be serializable if non-scalar.
+	 * @param mixed     $meta_type  One of 'user', 'post' or 'term'.
+	 *
+	 * @return bool
+	 */
+	private function filter_update_meta( $check, $object_id, $meta_key, $meta_value, $meta_type ) {
+		if ( strpos( $meta_key, '_yoast_wpseo_' ) === 0 ) {
+			$key = str_replace( '_yoast_wpseo_', '', $meta_key );
+			if ( ! isset( $this->map_meta_indexable[ $key ] ) ) {
+				return $check;
+			}
+			$field             = $this->map_meta_indexable[ $key ];
+			$indexable         = $this->indexable_repository->find_by_id_and_type( $object_id, $meta_type, false );
```

Comment:
Passing `false` to the auto-create here means this can potentially return `false` instead of an indexable.

Given we want to be sure to store the data I'd recommend not passing the third argument so the indexable is auto-created.

Do note that even in that case it's still possible for `false` ( for indexables that should not be indexed ) or an indexable with the `post_status` `unindexed` to be created ( for indexables of objects that do not exist ).

```
if ( $indexable === false || $indexable->post_status === 'unindexed' ) {
    return false;
}
```